smartlearner/optimizers/adam.py

```diff
@@ -0,0 +1,43 @@
+from collections import OrderedDict
+
+import theano
+import theano.tensor as T
+import numpy as np
+
+from ..interfaces import Optimizer
+from ..utils import sharedX
+
+class Adam(Optimizer):
+    def __init__(self, loss, learning_rate=0.0001, b1=0.9, b2=0.999, epsilon=1e-8, lmbda=(1 - 1e-8)):
+        super().__init__(loss)
+        self.lr = learning_rate
+        self.b1 = b1
+        self.b2 = b2
+        self.eps = epsilon
+        self.lmbda = lmbda
+
+        self._updates = OrderedDict()
+        self.t = sharedX(0, name='t')
+        self.mts = []
+        self.vts = []
+
+        for i, param in enumerate(loss.gradients):  # iterates on the parameters key
+            self.mts.append(sharedX(np.zeros_like(param.get_value()), name='mt{}'.format(i)))
```

Comment:
Oh, my mistake, I was thinking of `MonitorVariable`. Ok, I would still prefer having a meaningful name, even it requires to do the same tricks as in `MonitorVariable` (i.e. use `auto_name` if `name` is `None`).